src/main.js

```diff
@@ -0,0 +1,27 @@
+const testEnv = 'DATABASE_URL\nADMIN_EMAIL\nDEBUG';
+const alphanumericThatDoesNotStartWithDigit = /^[A-Z_][0-9A-Z_]*$/;
+
+//function that checks if the env value is valid
+//value cannot start with a digit and must only contain alphanumeric characters or an underscore
+function checkValidationOfValues(envSpecString){
+  let envValues = envSpecString.split('\n');//split lines based on \n character
+  envValues = envValues.filter(element =>{
+    if(element.match(alphanumericThatDoesNotStartWithDigit)){//keep valid values
+      return element;
+    }
+  })
+  return envValues;
+}
+
+//function that returns the HTML code as a string
+function outputHTML(envValues){
+  let outputString = "";
```

Comment:
Let's try achieving the same result without any side-effects. Side effects, like modifying variables outside of scope (e.g. `outputString` inside another function) are not a good practice, as it can lead to obscure and almost "undebuggable" situations.

---

## Proposal

Would something like that work?

```javascript
const output = envValues.map(value => (
    `<label for="env_spec_${value.toLowerCase()}">${value}</label>\n` +
    `<input id="env_spec_${value.toLowerCase()}" name="${value.toLowerCase()}" />\n`
)).join('\n')
```

Take a look at how [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and [`join`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) work.